### PR TITLE
[feat] Add JoyAI-Echo multi-shot audio-video generation pipeline

### LINKED ISSUE #13909: [New Pipeline/Model] Add JoyAI-Echo multi-shot audio-video generation pipeline
### Model/Pipeline/Scheduler description

**We are the JoyAI Team (JD.com)**, proposing the integration of JoyAI-Echo into Diffusers.

**JoyAI-Echo** is a unified framework for long-form audio-visual generation, designed to support minute-level video creation with synchronized audio, strong temporal consistency, and real-time interaction.

Key innovations:
- **Cross-modal audio-visual memory bank**: preserves character appearance and voice timbre across long sequences (up to minutes)
- **DMD-distilled few-step inference**: ~7.5× faster than baseline while improving alignment and visual quality
- **Joint audio-video generation**: a single pipeline produces synchronized video and audio
- **Multi-shot story generation**: generates coherent sequences of shots from prompt lists

The architecture builds on LTX-2 and adds the JoyAI-Echo DMD denoising schedule plus a paired audio-video memory bank for cross-shot consistency.

### Open source status

- [x] The model implementation is available.
- [x] The model weights are available (Only relevant if addition is not a scheduler).

### Provide useful links for the implementation

- GitHub Repository: https://github.com/jd-opensource/JoyAI-Echo
- HuggingFace Weights: https://huggingface.co/jdopensource/JoyAI-Echo
- Diffusers implementation PR: (will link after PR is created)

### Additional context

We (JoyAI Team) previously contributed JoyAI-Image-Edit to Diffusers (PR #13444, merged). This follows the same pattern — official team providing a complete, tested implementation.

### PR DESCRIPTION
**Pull request by sjq66**
## What does this PR do?

**We are the JoyAI Team (JD.com)**, and this is the Diffusers implementation for the JoyAI-Echo model.

- GitHub Repository: https://github.com/jd-opensource/JoyAI-Echo
- HuggingFace Weights: https://huggingface.co/jdopensource/JoyAI-Echo

Fixes #13909

## Model Overview

JoyAI-Echo is a unified framework for long-form audio-visual generation that supports minute-level multi-shot video creation with synchronized audio, strong temporal consistency, and real-time interaction.

### Key Features
- 🎞️ **Minute-level multi-shot stories**: generate a sequence of coherent shots from a list of prompts
- ⚡ **DMD-distilled few-step inference**: ~7.5× faster than the original pipeline
- 🔊 **Joint audio-video generation**: one pipeline produces synchronized video and audio
- 🧠 **Paired cross-modal memory bank**: conditions each new shot on prior visual identity and voice context for story-level consistency

### New files added:
- `src/diffusers/models/transformers/transformer_joyai_echo.py` — `JoyAIEchoTransformer3DModel`, extends LTX2VideoTransformer3DModel with memory mask support for multi-shot generation
- `src/diffusers/pipelines/joyai_echo/pipeline_joyai_echo.py` — `JoyAIEchoPipeline`, multi-shot pipeline with paired audio-video memory bank
- `src/diffusers/pipelines/joyai_echo/pipeline_joyai_echo_original_checkpoint.py` — `JoyAIEchoOriginalCheckpointPipeline` for loading original checkpoints
- `src/diffusers/pipelines/joyai_echo/pipeline_output.py` — Output dataclasses
- `docs/source/en/api/pipelines/joyai_echo.md` — API documentation
- `tests/pipelines/joyai_echo/test_joyai_echo.py` — Pipeline tests

### Usage example:

```python
import torch
from diffusers import JoyAIEchoPipeline
from diffusers.utils import encode_video

pipe = JoyAIEchoPipeline.from_pretrained("jdopensource/JoyAI-Echo", torch_dtype=torch.bfloat16)
pipe.enable_model_cpu_offload()

output = pipe(
    [
        "A cinematic opening shot of the protagonist entering a quiet train station.",
        "The same protagonist speaks softly while the camera follows through the platform.",
    ],
    height=736,
    width=1280,
    num_frames=241,
    frame_rate=25.0,
)

for i, (frames, audio) in enumerate(zip(output.frames, output.audio)):
    encode_video(frames[0], fps=25, audio=audio[0].float().cpu(), output_path=f"shot_{i:03d}.mp4")
```

## Tests

```
$ python -m pytest tests/pipelines/joyai_echo/ -v
tests/pipelines/joyai_echo/test_joyai_echo.py::JoyAIEchoPipelineFastTests::test_multishot_memory_prefix PASSED [100%]
======================== 1 passed in 6.54s =========================
```

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case. → #13909
- [x] Did you make sure to update the documentation with your changes? Here are the [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

- Pipelines and pipeline callbacks: @yiyixuxu and @asomoza
- Models: @dg845 and @a-r-r-o-w

### CODE REVIEW

**Comment by yiyixuxu** [comment]
@askserge can you do a initial review here?

**Comment by yiyixuxu** [comment]
@claude can you do a review here?

**Comment by sayakpaul** [comment]
Closing in favor of https://github.com/huggingface/diffusers/pull/14696
